src/lib/co2js/index.ts

```diff
@@ -68,55 +71,52 @@ export const Co2js = (globalConfig?: ConfigParams): PluginInterface => {
     return paramType[inputWithConfig.type]();
   };
 
+  const bytesSchema = z.object({
+    'network/data/bytes': z.number().optional(),
+    'network/data': z.number().optional(),
+  });
+  const greenWebHostSchema = z.object({
+    'green-web-host': z.boolean().optional(),
+  });
+  const globalConfigSchema = z.object({
+    options: z
+      .object({
+        dataReloadRatio: z.number().min(0).max(1).optional(),
+        firstVisitPercentage: z.number().min(0).max(1).optional(),
+        returnVisitPercentage: z.number().min(0).max(1).optional(),
+        gridIntensity: z
+          .object({
+            device: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+            dataCenter: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+            networks: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+          })
+          .optional(),
+      })
+      .optional(),
+  });
+
   /**
-   * Validates input parameters.
+   * Validates input and config parameters.
    */
   const validateInput = (input: PluginParams) => {
-    const schema = z
-      .object({
-        'network/data/bytes': z.number(),
-        'network/data': z.number(),
-      })
-      .partial()
+    const inputSchema = bytesSchema
+      .merge(greenWebHostSchema)
```

Comment:
Are you sure? This would be sad, because it wouldn't support my use case. I wrote a plugin to get the green-web-host parameter separately and feed it into this one. This doesn't work if it is a config option.

Reply:
We assumed this parameter was constant at the level of a node in the manifest but if that's not the case I think we can probably flex on whether it is in config or inputs. @manushak can either confirm or push back.

Also, just because there are times we don't want to merge something into our version of a plugin, doesn't mean you can't host your preferred version elsewhere. The ideal outcome for us is to have many people all maintaining version so the plugins that work for their specific use cases in many repositories, and we just maintain a minimal "standard lib" of essentials. We want the plugin ecoystem to be as decentralized as possible so we are strongly in favour of  people forking and modifying our plugins even if the changes don't get upstreamed.

Reply:
Hi @jmcook1186, thanks for your explanation. Makes sense to me.

My reasoning is that it is more convenient to have one standard version of a plugin that is allowing for a broad spectrum of use cases, than having several slightly modified versions of the same plugin throughout the ecosystem where people have to understand what the differences are etc. That might be confusing and a bit frustrating. 
A plugin that is as general as possible makes the latter case less likely. So I think there is value in making a plugin as versatile as possible.

But if you don't want to include it, of course, I will just create a variant of it.